example/app/utils/helper.ts

```diff
@@ -256,16 +276,20 @@ function getInputProps(name: string, def: ZodTypeAny): InputPropType {
   } else if (def instanceof ZodDate) {
     type = 'date'
   } else if (def instanceof ZodArray) {
-    return getInputProps(name, def.element)
+    return getInputProps(name, def.element, optional)
   } else if (def instanceof ZodOptional) {
-    return getInputProps(name, def.unwrap())
+    return getInputProps(name, def.unwrap(), true)
+  } else if (def instanceof ZodEffects) {
+    return getInputProps(name, def._def.schema, optional)
+  } else if (def instanceof ZodPipeline) {
+    return getInputProps(name, def._def.out, optional)
   }
 
   let inputProps: InputPropType = {
     name,
     type,
   }
-  if (!def.isOptional()) inputProps.required = true
+  if (!(def.isOptional() || optional)) inputProps.required = true
```

Comment:
The "optionality" of a type seemed to get lost when introspecting its inner type (which I think explains the discrepancy in the tests where we're asserting on the generated props for the optional `d` field from the base test schema but the assertion includes `required: true`).